### PR TITLE
Add pred_transform_function in Accuracy

### PR DESCRIPTION
**Pull request by ammarchalifah**
The motivation behind this pull request is to add flexibility in processing model output before calculating metric from it. One practical use case is for classification problem for ordered ordinal classes, which is easier to frame as a regression model. However, regression outputs a continuous-valued variable, therefore cannot be used to compute accuracy. In the prediction setting, we can clip and round the regression output to obtain integer i.e. the classification result. By adding pred_transform_function to accuracy, user can tweak model output and monitor their model better-suited to their needs.

### CODE REVIEW

**Comment by tomerk** [comment]
My initial thinking is the team would probably prefer to not further expand the API when there's already several existing ways to manipulate desired metrics computation.

https://keras.io/api/metrics/

E.g. Several ways of specifying custom metrics which are free to wrap other metrics, several variations of add_metrics, multi-output models that compute different metrics on different outputs, custom training steps that modify details of metrics computation, etc.

We'll discuss this at our sync though.

**Comment by ammarchalifah** [comment]
Thank you for your response, definitely waiting for updates after the sync

**Comment by tomerk** [comment]
Hi @ammarchalifah , we concluded that our api surface is too expansive as-is and so we unfortunately can't add this argument. You should be able to subclass the accuracy metric and override update_state (and call the super method in there) to make your own custom metric that has this functionality.
